test/unit/JsonService.test.ts

```diff
@@ -74,6 +84,28 @@ describe("JsonService", () => {
             expect(result).toEqual(json);
         });
 
+        it("should reject promise when http response is 200 and json is not able to parse", async () => {
+            // arrange
+            fetchMock.mockResolvedValue({
+                status: 200,
+                headers: new Headers({
+                    "Content-Type": "application/json"
+                }),
+                json: () => Promise.reject(new SyntaxError("Unexpected token a in JSON"))
+            });
+
+            try {
+                // act
+                await subject.getJson("http://test");
+                fail("should not come here");
+            }
+            catch (err) {
+                // assert
+                expect(err).toBeInstanceOf(SyntaxError);
+                expect((err as SyntaxError).message).toContain("Unexpected token a in JSON");
+            }
```

Comment:
Same here, you can reduce this to

```suggestion
            await expect(subject.getJson("http://test")).rejects.toThrow(new SyntaxError("Unexpected token a in JSON"));
```